### ISSUE TITLE
Project contribution

### ISSUE THREAD

**Issue by diyor28** [cloudblue/django-rql#48]
Hi, I use django-rql on daily basis and I would like to contribute to the project. Please let me know what I can work on

**Comment by maxipavlovic** [comment]
@diyor28 great to hear that! Just for our curiosity, would you like to share your use cases?

About contribution, thank you for the intention! Currently, we are in the process of creation of [CLA](https://en.wikipedia.org/wiki/Contributor_License_Agreement#:~:text=A%20Contributor%20License%20Agreement%20(CLA,under%20an%20open%20source%20license.) for individual contributors, and I will ping you when we are done.

By the way, you can check CloudBlue vacancies [here](https://careers.ingrammicro.com/search/cloudblue/jobs) :)

**Comment by diyor28** [comment]
Sorry for the late reply, missed the notification. Thank you for the invitation to work at CloudBlue! Unfortunately, I'm already an Epam employee. 

Regarding the use cases, I usually use it for my personal projects where I need to build a flexible API for a dashboard or admin panel with Django, Django Rest Framework, and Postgres as the backend. The way I found django-rql is quite interesting. I was working with [feathersjs](https://docs.feathersjs.com/api/databases/querying.html), [Feathers-Vuex](https://vuex.feathersjs.com/) and I loved the flexing and versatile query language that feathersjs provides, so I wanted to find a similar tool for django. And my search lead me to django-rql.

By the way, I created a typescript version of javascript-rql, since my contribution issue was missed for some reason. You can check it out [here](https://www.npmjs.com/package/js-rql).

**Comment by maxipavlovic** [comment]
Thx for the explanation! Will keep in mind [featherjs](https://docs.feathersjs.com/api/databases/querying.html).

Couldn't check TS version as https://github.com/diyor28/js-rql from npm page leads to 404.

Sorry to hear about missing your issue. 
@borzovplus @AntonHinz please, help here https://github.com/cloudblue/javascript-rql/issues/10 :)

**Comment by diyor28** [comment]
My bad,  https://github.com/diyor28/js-rql is now a valid repo.

**Comment by diyor28** [comment]
Regarding django-rql for python, I noticed that it does not support querying by ForeignKey directly or it might just be that I missed it in the docs

**Comment by maxipavlovic** [comment]
You mean something like this?

```python

class Model:
  group = FK()

class Filters:
  MODEL = Model
  FILTERS = (
    {
        'filter': 'group.id',
        'source': 'group_id',
    },
```

I guess, that it's a great note for index optimization purposes, which is not supported at this moment.

**Comment by diyor28** [comment]
Yeah, exactly. I believe it is a  quite common use case where you might need to query records belonging to a certain user. I would love to bring support for that.

**Comment by maxipavlovic** [comment]
@diyor28 CLA is ready and integrated. Waiting for your contributions :)

**Comment by diyor28** [comment]
@maxipavlovic Happy to hear that! What part of it do you think I could start with?

**Comment by maxipavlovic** [comment]
Let's start with direct support of FKs as you proposed :)

**Comment by diyor28** [comment]
@maxipavlovic Hi, I would love to bring support for that, but I am incredibly busy right now. As soon as I get a few hours of free time I will get to work on this. Thank you for your patience.
